src/index.ts

```diff
@@ -78,19 +73,16 @@ async function fetchAndTrace(url: string, init: RequestInit): Promise<Response>
     _url = new URL(url);
   } catch (error) {}
   if (sentry && _url && (tracingOrigins.includes(_url.origin) || tracingPaths.includes(_url.pathname))) {
-    const transaction = sentry.startTransaction({
-      name: url,
-    });
-    const span = transaction.startChild({
-      op: "http",
-    }); // This function returns a Span
-
-    const response = await fetch(url, init);
-    span.finish(); // Remember that only finished spans will be sent with the transaction
-
-    transaction.finish(); // Finishing the transaction will send it to Sentry
-
-    return response;
+    sentry.startSpan(
+      {
+        name: url,
+        op: "http",
+      },
+      async () => {
+        const response = await fetch(url, init);
+        return response;
```

Comment:
we need to return response on original call and not inside the span

Reply:
you need to return the result from sentry.startSpan

Reply:
op should be http.client

Reply:
https://docs.sentry.io/platforms/javascript/tracing/instrumentation/custom-instrumentation/requests-module/